tests/pyfunc/test_pyfunc_schema_enforcement.py

```diff
@@ -2559,6 +2559,124 @@ def predict(self, context, model_input, params=None):
+                    {},
+                    {"c": {}},
+                ]
+            },
+            Schema([ColSpec(Map(value_type=Map(value_type=DataType.long)), name="nested_map")]),
+        ),
+        (
+            {
+                "array_in_map": [
+                    {"a": [1, 2, 3], "b": [4, 5]},
+                    {},
+                    {"c": []},
+                ]
+            },
+            Schema([ColSpec(Map(value_type=Array(dtype=DataType.long)), name="array_in_map")]),
+        ),
+        (
+            {
+                "object_in_map": [
+                    {"a": {"key1": "a1", "key2": 1}, "b": {"key1": "b1"}},
+                    {},
+                    {"c": {"key1": "c1"}},
+                ]
+            },
+            Schema(
+                [
+                    ColSpec(
+                        Map(
+                            value_type=Object(
+                                [
+                                    Property("key1", DataType.string),
+                                    Property("key2", DataType.long, required=False),
+                                ]
+                            )
+                        ),
+                        name="object_in_map",
+                    )
+                ]
+            ),
+        ),
+        (
+            {
+                "map_in_array": [
+                    [{"a": 3, "b": 4}, {"c": 5}],
+                    [],
+                    [{"d": 6}],
+                ]
+            },
+            Schema([ColSpec(Array(dtype=Map(value_type=DataType.long)), name="map_in_array")]),
+        ),
+        (
+            {
+                "map_in_object": [
+                    {"key1": {"a": 3, "b": 4}, "key2": {"c": 5}},
+                    {"key1": {"d": 6}},
+                ]
+            },
+            Schema(
+                [
+                    ColSpec(
+                        Object(
+                            [
+                                Property("key1", Map(value_type=DataType.long)),
+                                Property("key2", Map(value_type=DataType.long), required=False),
+                            ]
+                        ),
+                        name="map_in_object",
+                    )
+                ]
+            ),
+        ),
+    ],
+)
+def test_pyfunc_model_schema_enforcement_map_type(data, schema):
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input, params=None):
+            return model_input
+
+    df = pd.DataFrame.from_records(data)
+
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            python_model=MyModel(),
+            artifact_path="test_model",
+            signature=ModelSignature(inputs=schema, outputs=schema),
+        )
+    loaded_model = mlflow.pyfunc.load_model(model_info.model_uri)
+    prediction = loaded_model.predict(df)
+    pd.testing.assert_frame_equal(prediction, df)
```

Comment:
I think we need follow-up work to make model serving works :) And also databricks model serving